Examples/pixmaps.lisp

```diff
@@ -0,0 +1,172 @@
+;;; sheet and to obtain both sheet and coordinates of the blank-area object.
+
+(defun draw-grid (frame medium)
+  (let ((inks (make-contrasting-inks 4)))
+    (when (null frame)
+      (setf inks (reverse inks)))
+    (draw-rectangle* medium 10 10 90 90 :ink (elt inks 0))
+    (draw-rectangle* medium 20 20 80 80 :ink (elt inks 1))
+    (draw-rectangle* medium 30 30 70 70 :ink (elt inks 2))
+    (draw-rectangle* medium 40 40 60 60 :ink (elt inks 3))))
+
+(defclass pixmap-sheet (clim-stream-pane)
+  ((pixmap :accessor sheet-pixmap :initform nil)))
+
+(defun reset-pixmap (sheet)
+  (alexandria:when-let ((pixmap (sheet-pixmap sheet)))
+    (deallocate-pixmap pixmap))
+  (setf (sheet-pixmap sheet)
+        (with-output-to-pixmap (medium sheet :width 100 :height  100)
+          (medium-clear-area medium 0 0 100 100)
+          (draw-grid nil medium)))
+  (repaint-sheet sheet +everywhere+))
+
+(defmethod note-sheet-grafted :after ((sheet pixmap-sheet))
+  (reset-pixmap sheet))
+
+(defmethod note-sheet-degrafted :before ((sheet pixmap-sheet))
+  (alexandria:when-let ((pixmap (sheet-pixmap sheet)))
+    (deallocate-pixmap pixmap))
+  (setf (sheet-pixmap sheet) nil))
+
+(defmethod repaint-sheet ((sheet pixmap-sheet) region)
+  (with-sheet-medium (medium sheet)
+    (medium-copy-area (sheet-pixmap sheet) 0 0 100 100 medium 0 0)))
+
+(define-application-frame pixmaps ()
+  ()
+  (:panes (app1 :application :display-time nil :display-function 'draw-grid
+                             :scroll-bars nil :borders nil
+                             :min-width 100 :max-width 100
+                             :min-height 100 :max-height 100)
+          (app2 :clim-stream-pane
+                :min-width 100 :max-width 100
+                :min-height 100 :max-height 100)
+          (pix1 pixmap-sheet :min-width 100 :max-width 100
+                             :min-height 100 :max-height 100)
+          (pix2 pixmap-sheet :min-width 100 :max-width 100
+                             :min-height 100 :max-height 100))
+  (:layouts (default (horizontally ()
+                       (labelling (:label "Source") app1)
+                       (labelling (:label "Target") app2)
+                       (labelling (:label "Pixmap 1") pix1)
+                       (labelling (:label "Pixmap 2") pix2))))
+  (:menu-bar nil))
+
+(defun region-feedback (frame presentation stream x0 y0 x y state)
+  (declare (ignore frame presentation))
+  (with-identity-transformation (stream)
+    (ecase state
+      (:highlight
+       (with-output-recording-options (stream :record nil)
+         (draw-rectangle* stream x0 y0 x y :filled nil
+                                           :line-thickness 3
+                                           :line-dashes t)))
+      (:unhighlight
+       (let ((region (make-rectangle* (- (min x0 x) 2) (- (min y0 y) 2)
+                                      (+ (max x0 x) 2) (+ (max y0 y) 2))))
```

Comment:
Could these be passed to `draw-rectangle*` directly?

Reply:
they could, but we call repaint-sheet with the region argument anyway, so I think that this is cleaner to cons the region once and use the result.